File: php/src/Services/UserAssetsService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;
use CCPayment\APIError;
use GuzzleHttp\Exception\GuzzleException;

class UserAssetsService
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserCoinAssetList(string $userId): array
    {
        return $this->client->post('/getUserCoinAssetList', ['userId' => $userId]);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserCoinAsset(string $userId, int $coinId): array
    {
        return $this->client->post('/getUserCoinAsset', ['userId' => $userId, 'coinId' => $coinId]);
    }
}

File: ruby/lib/ccpayment/services/user_assets_service.rb
module CCPayment
  module Services
    class UserAssetsService
      def initialize(client)
        @client = client
      end

      def get_user_coin_asset_list(user_id)
        @client.post('/getUserCoinAssetList', { userId: user_id })
      end

      def get_user_coin_asset(user_id, coin_id)
        @client.post('/getUserCoinAsset', { userId: user_id, coinId: coin_id })
      end
    end
  end
end
